File: app/src/main/java/ru/systempla/weatherapp/mvp/model/api/IDataSource.java
package ru.systempla.weatherapp.mvp.model.api;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;
import ru.systempla.weatherapp.mvp.model.entity.ForecastRequestRestModel;
import ru.systempla.weatherapp.mvp.model.entity.UVIRequestRestModel;
import ru.systempla.weatherapp.mvp.model.entity.WeatherRequestRestModel;


public interface IDataSource {
    @GET("data/2.5/weather")
    Single<WeatherRequestRestModel> loadWeather(@Query("q") String city,
                                                @Query("appid") String keyApi,
                                                @Query("units") String units,
                                                @Query("lang") String language);

    @GET("data/2.5/uvi")
    Single<UVIRequestRestModel> loadUVI(@Query("appid") String keyApi,
                                        @Query("lat") double latitude,
                                        @Query("lon") double longitude);

    @GET("data/2.5/forecast")
    Single<ForecastRequestRestModel> loadForecast(@Query("q") String city,
                                                  @Query("appid") String keyApi,
                                                  @Query("units") String units,
                                                  @Query("lang") String language);
}

File: app/src/main/java/ru/systempla/weatherapp/mvp/model/api/IDataSource.kt
package ru.systempla.weatherapp.mvp.model.api

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import ru.systempla.weatherapp.mvp.model.entity.ForecastRequestRestModel
import ru.systempla.weatherapp.mvp.model.entity.UVIRequestRestModel
import ru.systempla.weatherapp.mvp.model.entity.WeatherRequestRestModel

interface IDataSource {
    @GET("data/2.5/weather")
    fun loadWeather(@Query("q") city: String,
                    @Query("appid") keyApi: String,
                    @Query("units") units: String,
                    @Query("lang") language: String): Single<WeatherRequestRestModel>

    @GET("data/2.5/uvi")
    fun loadUVI(@Query("appid") keyApi: String,
                @Query("lat") latitude: Double,
                @Query("lon") longitude: Double): Single<UVIRequestRestModel>

    @GET("data/2.5/forecast")
    fun loadForecast(@Query("q") city: String,
                     @Query("appid") keyApi: String,
                     @Query("units") units: String,
                     @Query("lang") language: String): Single<ForecastRequestRestModel>
}